docs/docs/50-user-guide/60-reference-docs/30-promotion-steps/jira.md

```diff
@@ -0,0 +1,651 @@
+                      - type: text
+                        text: " "
+                  - type: heading
+                    attrs:
+                      level: 3
+                    content:
+                      - type: text
+                        text: "Automated deployment ticket for release "
+                      - type: text
+                        text: "${{ imageFrom(vars.imageRepo).Tag }}"
+                        marks:
+                          - type: code
+                  - type: paragraph
+                    content:
+                      - type: text
+                        text: "Image:"
+                        marks:
+                          - type: strong
+                      - type: text
+                        text: " "
+                      - type: text
+                        text: "${{ imageFrom(vars.imageRepo).RepoURL }}:${{ imageFrom(vars.imageRepo).Tag }}"
+                        marks:
+                          - type: code
+                  - type: paragraph
+                    content:
+                      - type: text
+                        text: "Project:"
+                        marks:
+                          - type: strong
+                      - type: text
+                        text: " "
+                      - type: text
+                        text: "${{ ctx.project }}"
+                        marks:
+                          - type: code
+              labels:
+                - "automated-deployment"
+                - "env-${{ ctx.stage.name }}"
+                - "release-${{ imageFrom(vars.imageRepo).Tag }}"
+                - "project-${{ ctx.project }}"
+
+        # Update application
+        - as: update-app
+          uses: argocd-update
+          config:
+            apps:
+              - name: test-app
+                namespace: argocd
+                sources:
+                  - repoURL: https://github.com/company/app-config.git
+                    kustomize:
+                      images:
+                        - repoURL: public.ecr.aws/nginx/nginx
+                          tag: ${{ imageFrom("public.ecr.aws/nginx/nginx").Tag }}
+
+        # Add progress comment
+        - as: comment-on-ticket
+          uses: jira
+          config:
+            credentials:
+              secretName: jira
+            commentOnIssue:
+              issueKey: ${{ outputs['create-deployment-ticket'].key }}
+              body: "Release ${{ imageFrom(vars.imageRepo).Tag }} has been promoted to ${{ ctx.stage.name }} environment at ${{ ctx.promotion.creationTimestamp }}. Freight: ${{ ctx.targetFreight.name }}. Ready for testing."
+
+        # Cleanup on failure
+        - as: on-failure-cleanup-issue
+          uses: jira
+          if: ${{ failure() }}
+          config:
+            credentials:
+              secretName: jira
+            deleteIssue:
+              issueKey: ${{ outputs['create-deployment-ticket'].key }}
+              deleteSubtasks: true
```

Comment:
Seeing it in this context, I think updating the [standalone example I commented on](https://github.com/akuity/kargo/pull/4497/files#r2178488575) to more closely resemble this might resolve that comment.

Reply:
Although I wonder if this is a realistic pattern...

Assuming many users will emulate what they see here, is deleting the Jira issue really a good way to respond to a promotion failure?

I see other Stages' promotion processes have similar cleanup steps. Do we really want to have an issue tracking successful promotion through several Stages and then throw all that information away because of a hiccup farther downstream?

Reply:
Nvm... I see now that subsequent Stages are cleaning up a _comment_ on failure.